bin/pulsar

```diff
@@ -307,6 +307,9 @@ if [[ -z "$IS_JAVA_8" ]]; then
   OPTS="$OPTS --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED"
 fi
 
+# Specify default nar tmp dir
+OPTS="$OPTS -Dnar.extraction.tmpdir=$PULSAR_HOME/tmp"
```

Comment:
we should make it overridable by a ENV variable.
for instance in the docker images it is better to not let the nar files be unpacked in /pulsar/tmp
the same applies for regular non-docker/non-k8s installations, the PULSAR_HOME directory contains "code" and "configuration" it shouldn't contain an arbitrary amount of data)

Reply:
@eolivelli Relate to PR https://github.com/apache/pulsar/pull/15979, we define `nar.extraction.tmpdir` ， the dir only contain some class file which extraction from nar package, use the dir in pulsar can avoid those .class file be clean by os in some linux os;
Diff linux os has diff behaviours for the /tmp dir clean, Debian will clean when the boot , but as Centos or RedHat will auto clean the file in the /tmp dir when 7 days or 30 days the file not be used, In this way, the .class file will loss.
In our case, we add the KOP protocol, after few days when we want use the KOP protocol, we find some .class have been cleaned by os.

Reply:
I agree that this option should be configurable.

Otherwise, you can actually run `OPTS=-Dnar.extraction.tmpdir=/path/to/pulsar/tmp bin/pulsar ...` and achieve the same function.